### PR TITLE
Update technical debt document

### PR DESCRIPTION
**Pull request by harrisj**
Revised to match the current file layout. Could probably use some more work, but it's getting there.

### CODE REVIEW

File: docs/technical_debt.md [hunk 1]
```diff
@@ -14,8 +11,9 @@ branching nests of conditionals. We're trying a few techniques on this
 project to forestall that that could be summarized with the following principles:
 
 1. Long model classes are bad
-2. NullObject is better than nil
-3. View logic should not be in the templates
+2. Composition/delegation is better than inheritance
```
Comment (jessieay) [inline docs/technical_debt.md]: we do have some inheritance still, eg: `BaseRules`. I think inheritance is fine as long as it is "shallow and wide"

File: docs/technical_debt.md [hunk 2]
```diff
@@ -14,8 +11,9 @@ branching nests of conditionals. We're trying a few techniques on this
...
+3. NullObject is better than nil
```
Comment (jessieay) [inline docs/technical_debt.md]: ❤️

**Comment by jessieay** [comment]
This is great! 👍

File: docs/technical_debt.md (hunk 1 above)
Comment (harrisj) [inline docs/technical_debt.md]: True. I feel like maybe I should elaborate with the Rules object a little

**Comment by harrisj** [comment]
Hold on, let me fix all the URL references

**Comment by jessieay** [comment]
@harrisj let me know when this is ready for re-review! (you mentioned something about wanting to add docs on validator)

**Comment by harrisj** [comment]
Needs to add in BidValidator as part of the user places bid walkthrough

**Comment by harrisj** [comment]
Is there a reason this was assigned to me? It's ready for review

**Comment by jessieay** [comment]
@harrisj cool, will re-review! I assigned you because I thought it wasn't ready - my bad!

**Pull request merged**